api/cas-server-core-api-events/src/main/java/org/apereo/cas/support/events/ticket/CasTicketGrantingTicketDestroyedEvent.java

```diff
@@ -30,5 +33,6 @@ public class CasTicketGrantingTicketDestroyedEvent extends AbstractCasTicketGran
      */
     public CasTicketGrantingTicketDestroyedEvent(final Object source, final TicketGrantingTicket ticket) {
         super(source, ticket);
+        this.clientInfo = ClientInfoHolder.getClientInfo();
```

Comment:
There is a fair amount of duplication here. I think ClientInfo is something that all events could benefit from, and so it likely should be moved upstream even further. Declaring the same definition 4-5 times is generally a good indication to refactor.

Reply:
I agree.  Should there be a class between AbstractCasTicketGrantingTicketEvent and these events, or just added to AbstractCasTicketGrantingTicketEvent?  I haven't inspected all the implementations of AbstractCasTicketGrantingTicketEvent, and I'm not sure its appropriate to add it to there.

Reply:
Or even AbstractCasEvent?